nodes/src/nodes/dataset_cobalt/dataset_loader.py

```
@@ -0,0 +1,212 @@
+        questions = []
+        for item in items:
+            text = item.get('input', item.get('text', item.get('question', '')))
+            metadata = {
+                'expected': item.get('expected', item.get('output', '')),
+                'dataset_id': item.get('id', ''),
+                'cobalt_source': True,
+            }
+            # Preserve any extra fields from the original item as metadata
+            for key, value in item.items():
+                if key not in ('input', 'text', 'question', 'expected', 'output', 'id'):
+                    metadata[key] = value
```

Ruff (0.15.7)

[warning] Use a dictionary comprehension instead of a for-loop

(PERF403)